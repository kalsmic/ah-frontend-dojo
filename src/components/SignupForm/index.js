// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

// styles
import 'components/SignupForm/SignupForm.scss';

// components
import Button from 'components/Button';
import InputBox from 'components/InputBox';
import ModalBox from 'components/ModalBox';

//  string constants
import { SIGN_UP_STRING } from 'constants/constantStrings';

/**
/**
 * This is a function which return the SignupForm component
 * @param {*} n - mixed parameter types*
 * @return {component}
 * @example
 * foo = (props) => {}
 */
const SignupForm = (props) => {
  const {
    onChangeHandler, email, username, password,
    confirmPassword, onSubmitHandler, emailError, usernameError,
    passwordError, confirmPasswordError, signupMessageState, isSigningUp, backdropId,
  } = props;

  return (
    // Signup form enclosed inside the modal
    <ModalBox show title={SIGN_UP_STRING} {...props} backdropId={backdropId}>
      <div>
        <form className="signupForm" id="signupForm">
          <InputBox
            name="email"
            inputType="email"
            handleChange={onChangeHandler}
            placeholder="email"
            inputClass="signupForm__input"
            value={email}
          >
            <br />
            <span id="signupForm__input__error__email" className="signupForm__input__error">
              <small>{emailError}</small>
            </span>
          </InputBox>
          <InputBox
            name="username"
            inputType="text"
            handleChange={onChangeHandler}
            placeholder="username"
            inputClass="signupForm__input"
            value={username}
          >
            <br />
            <span id="signupForm__input__error__username" className="signupForm__input__error">
              <small>{usernameError}</small>
            </span>
          </InputBox>
          <InputBox
            name="password"
            inputType="password"
            handleChange={onChangeHandler}
            placeholder="password"
            inputClass="signupForm__input"
            value={password}
          >
            <br />
            <span id="signupForm__input__error__password" className="signupForm__input__error">
              <small>{passwordError}</small>
            </span>
          </InputBox>
          <InputBox
            name="confirmPassword"
            inputType="password"
            handleChange={onChangeHandler}
            placeholder="confirm password"
            inputClass="signupForm__input"
            value={confirmPassword}
          >
            <br />
            <span id="signupForm__input__error__confirmPassword" className="signupForm__input__error">
              <small>{confirmPasswordError}</small>
            </span>
          </InputBox>
          <br />
          <span className="signupForm__signUpMessage">{signupMessageState}</span>
          {isSigningUp && <div className="signupForm__loader" />}
          <Button
            id="signupForm__button"
            btnClass="signupForm__button"
            btnName={SIGN_UP_STRING}
            btnEvent={onSubmitHandler}
          />
        </form>
      </div>
    </ModalBox>
  );
};

// prop types validations
SignupForm.propTypes = {
  onChangeHandler: PropTypes.func,
  email: PropTypes.string,
  username: PropTypes.string,
  password: PropTypes.string,
  confirmPassword: PropTypes.string,
  onSubmitHandler: PropTypes.func,
  emailError: PropTypes.string,
  usernameError: PropTypes.string,
  passwordError: PropTypes.string,
  confirmPasswordError: PropTypes.string,
  signupMessageState: PropTypes.string,
  isSigningUp: PropTypes.bool,
  backdropId: PropTypes.string,
};

// default props
SignupForm.defaultProps = {
  onChangeHandler: () => { },
  email: '',
  username: '',
  password: '',
  confirmPassword: '',
  onSubmitHandler: () => { },
  emailError: '',
  usernameError: '',
  passwordError: '',
  confirmPasswordError: '',
  signupMessageState: '',
  isSigningUp: false,
  backdropId: 'signupModal',
};

export default SignupForm;
