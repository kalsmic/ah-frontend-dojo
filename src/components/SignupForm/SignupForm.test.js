// react libraries
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';

// components
import SignupForm from 'components/SignupForm';

/**
 * This is a function which tests SignupForm component
 */
describe('SignupForm Component', () => {
  it('matches the snapshot', () => {
    const component = shallow(<SignupForm backdropId="signupModal" />);
    expect(component).toMatchSnapshot();
  });
});
