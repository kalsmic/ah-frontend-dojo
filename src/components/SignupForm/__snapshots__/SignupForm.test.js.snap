// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm Component matches the snapshot 1`] = `
<ModalBox
  backdropId="signupModal"
  confirmPassword=""
  confirmPasswordError=""
  email=""
  emailError=""
  isSigningUp={false}
  onChangeHandler={[Function]}
  onSubmitHandler={[Function]}
  password=""
  passwordError=""
  show={true}
  signupMessageState=""
  title="Sign up"
  username=""
  usernameError=""
>
  <div>
    <form
      className="signupForm"
      id="signupForm"
    >
      <InputBox
        handleChange={[Function]}
        inputClass="signupForm__input"
        inputType="email"
        name="email"
        placeholder="email"
        required={true}
        value=""
      >
        <br />
        <span
          className="signupForm__input__error"
          id="signupForm__input__error__email"
        >
          <small />
        </span>
      </InputBox>
      <InputBox
        handleChange={[Function]}
        inputClass="signupForm__input"
        inputType="text"
        name="username"
        placeholder="username"
        required={true}
        value=""
      >
        <br />
        <span
          className="signupForm__input__error"
          id="signupForm__input__error__username"
        >
          <small />
        </span>
      </InputBox>
      <InputBox
        handleChange={[Function]}
        inputClass="signupForm__input"
        inputType="password"
        name="password"
        placeholder="password"
        required={true}
        value=""
      >
        <br />
        <span
          className="signupForm__input__error"
          id="signupForm__input__error__password"
        >
          <small />
        </span>
      </InputBox>
      <InputBox
        handleChange={[Function]}
        inputClass="signupForm__input"
        inputType="password"
        name="confirmPassword"
        placeholder="confirm password"
        required={true}
        value=""
      >
        <br />
        <span
          className="signupForm__input__error"
          id="signupForm__input__error__confirmPassword"
        >
          <small />
        </span>
      </InputBox>
      <br />
      <span
        className="signupForm__signUpMessage"
      />
      <Button
        btnClass="signupForm__button"
        btnEvent={[Function]}
        btnName="Sign up"
        id="signupForm__button"
      />
    </form>
  </div>
</ModalBox>
`;
